analyzers/src/SonarAnalyzer.CSharp/Rules/ValuesUselesslyIncremented.cs

```diff
@@ -32,26 +32,21 @@ namespace SonarAnalyzer.Rules.CSharp
     [Rule(DiagnosticId)]
     public sealed class ValuesUselesslyIncremented : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S2123";
+        private const string DiagnosticId = "S2123";
         private const string MessageFormat = "Remove this {0} or correct the code not to waste it.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var increment = (PostfixUnaryExpressionSyntax)c.Node;
                     var symbol = c.SemanticModel.GetSymbolInfo(increment.Operand).Symbol;
-                    var localSymbol = symbol as ILocalSymbol;
-                    var parameterSymbol = symbol as IParameterSymbol;
 
-                    if (localSymbol == null &&
-                        (parameterSymbol == null || parameterSymbol.RefKind != RefKind.None))
+                    if (!(symbol is ILocalSymbol || symbol is IParameterSymbol {RefKind: RefKind.None}))
```

Comment:
nice! however, I find it easier to read it like:

```
if (symbol is not ILocalSymbol && symbol is not IParameterSymbol {RefKind: RefKind.None})
```

WDYT? The `!` is quite hidden and I initially didn't see it. Also, the above is clearer that it's an `and` condition.

Or even better, get rid of the early return (and negative logic) and add the rest of the function inside the if

```
if (symbol is ILocalSymbol || symbol is IParameterSymbol {RefKind: RefKind.None})
{
   // the logic here
}
```

Like this, it's crystal clear. Only one logic flow.